src/gui/plugins/playback_scrubber/PlaybackScrubber.cc

```diff
@@ -0,0 +1,235 @@
+      [this](const Entity &,
+           const components::LogPlaybackStatistics *_logStatComp)->bool
+      {
+        auto startSeconds =
+          _logStatComp->Data().start_time().sec();
+        auto startNanoseconds =
+          _logStatComp->Data().start_time().nsec();
+        auto endSeconds =
+          _logStatComp->Data().end_time().sec();
+        auto endNanoseconds =
+          _logStatComp->Data().end_time().nsec();
+        this->dataPtr->startTime =
+          math::secNsecToTimePoint(startSeconds, startNanoseconds);
+        this->dataPtr->endTime =
+          math::secNsecToTimePoint(endSeconds, endNanoseconds);
+        return true;
+      });
+  }
+
+  auto simTime = math::durationToSecNsec(_info.simTime);
+  this->dataPtr->currentTime =
+    math::secNsecToTimePoint(simTime.first, simTime.second);
+  this->dataPtr->progress = CalculateProgress();
+  this->dataPtr->paused = _info.paused;
+  this->newProgress();
+}
+
+/////////////////////////////////////////////////
+double PlaybackScrubber::Progress()
+{
+  return this->dataPtr->progress;
+}
+
+/////////////////////////////////////////////////
+QString PlaybackScrubber::StartTimeAsString()
+{
+  return QString::fromStdString(
+      math::timePointToString(this->dataPtr->startTime));
+}
+
+/////////////////////////////////////////////////
+QString PlaybackScrubber::EndTimeAsString()
+{
+  return QString::fromStdString(
+      math::timePointToString(this->dataPtr->endTime));
+}
+
+/////////////////////////////////////////////////
+QString PlaybackScrubber::CurrentTimeAsString()
+{
+  return QString::fromStdString(
+      math::timePointToString(this->dataPtr->currentTime));
+}
+
+/////////////////////////////////////////////////
+void PlaybackScrubber::OnTimeEntered(const QString &_time)
+{
+  std::string time = _time.toStdString();
+  std::chrono::system_clock::time_point enteredTime =
+    math::stringToTimePoint(time);
+  if (enteredTime == std::chrono::system_clock::from_time_t(-1))
+  {
+    ignwarn << "Invalid time entered. "
+      "The format is dd hh:mm:ss.nnn" << std::endl;
+    return;
+  }
+
+  // Check for time out of bounds
+  if (enteredTime < this->dataPtr->startTime)
+    enteredTime = this->dataPtr->startTime;
+  else if (enteredTime > this->dataPtr->endTime)
+    enteredTime = this->dataPtr->endTime;
+
+  auto pairTime = math::timePointToSecNsec(enteredTime);
+  const std::string topic = "/world/default/playback/control";
+  unsigned int timeout = 1000;
+  msgs::Boolean res;
+  bool result{false};
+  msgs::LogPlaybackControl playbackMsg;
+
+  // Set time and make request
+  playbackMsg.mutable_seek()->set_sec(pairTime.first);
+  playbackMsg.mutable_seek()->set_nsec(pairTime.second);
+  playbackMsg.set_pause(true);
+  this->dataPtr->node.Request(topic, playbackMsg, timeout, res, result);
+}
+
+/////////////////////////////////////////////////
+void PlaybackScrubber::OnDrag(double _value)
+{
+  const std::string topic = "/world/default/playback/control";
```

Comment:
Same thing about hardcoding the world name here

Reply:
Resolved e7a428e1e3fbd4c02f4f17a14817dcadf989b751